### DISCUSSION TITLE
L字コンスルーを破損してしまいました

### DISCUSSION THREAD

**Discussion by ghost** [Yowkees/keyball#290] [Q&A]
表題の通り、作成途中で誤ってL字コンスルーを逆側にハンダ付けしてしまい、
うまく取り外すことが出来ず破損させてしまいました。

ProMicroに付属していたピンヘッダを加工してみたのですが…
![image](https://github.com/Yowkees/keyball/assets/85214881/efe683bc-4950-4a15-bdd5-1cd776420b0a)

やはり…というか、トラックボールが動作しません。
LCDに表示される移動量も0のままです。
（キーについては全て正常に入力できます）

ただ、ボールを近づけると、それに呼応するようにセンサーが光ることは光ります。
![image](https://github.com/Yowkees/keyball/assets/85214881/dc150510-d1fe-4251-b4a5-c12d66fff626)
![image](https://github.com/Yowkees/keyball/assets/85214881/039e4260-0c80-4242-a552-e22b7cd1297d)
![image](https://github.com/Yowkees/keyball/assets/85214881/e3500be4-09d9-488f-b8b6-733a8bab97d9)
![image](https://github.com/Yowkees/keyball/assets/85214881/eee58e99-3b70-4876-95c7-5d47aaa8c73f)

工作初心者ということもあり、L字コンスルー部分の接触不良なのか、センサーが破損しているのか、
ハンダのつけ忘れなのか切り分けることもできず。

L字コンスルーがとても貴重な部品、、、ということを破損した後に認識し、青ざめております。。
大変恐縮ですが、よろしくお願いいたします。

**Comment by ghost** [comment]
切り分けできました。

やはり接触不良のようでした。
L字コンスルーを模したピンヘッダの角度を調整して、センサー基板のホールの縁に当たるようにしたら、
トラックボールが反応するようになりました。感動です。

ただ、ちょっとした振動で使えなくなったりするので、思い切ってハンダ付けしてしまうべきか
迷いどころです。

意外に収まりは悪くなさそうではあるのですが…。
![image](https://github.com/Yowkees/keyball/assets/85214881/c78b02c7-6753-45ef-b17c-5f08cf2ad2a4)

Aliexpressあたりを探してみても、L字コンスルーに相当する部品を見つけ出すことはできませんでした。
希少な部品とは理解しているのですが、、、どうにか分けていただくことはできないでしょうか。

はじめての自作キーボードづくりで、ここまで来るのに苦難の連続で何度もくじけそうになりましたが、、
トラックボールが動いてくれてホントに嬉しいです。
色々と工夫しつつメンテしながら、生涯愛用したいです。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。
L字7Pコンスルー単体で販売可能です。

やはり通常のピンヘッダーでは、完全に導通させるのは難しく、半田付けしてしまうとメンテナンスが難しくなるので、新しいコンスルーを使用することをお勧めします。

ご購入希望の場合は、送付先等お話ししたいので、下記よりご連絡ください。
https://shirogane-lab.net/inquiry

以上、よろしくお願いいたします。

**Reply by ghost** [reply]
ありがとうございます！
ご連絡させていただきます。
